tests/integration/test_authd/test_authd_ssl_certs.py

```diff
@@ -0,0 +1,223 @@
+
+type: integration
+
+brief: These tests will check if the 'wazuh-authd' daemon is able to handle secure connections using
+       the 'SSL' (Secure Socket Layer) protocol. The 'wazuh-authd' daemon can automatically add
+       a Wazuh agent to a Wazuh manager and provide the key to the agent.
+       It is used along with the 'agent-auth' application.
+
+components:
+    - authd
+
+targets:
+    - manager
+
+daemons:
+    - wazuh-authd
+    - wazuh-db
+    - wazuh-modulesd
+
+os_platform:
+    - linux
+
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - Debian Buster
+    - Red Hat 8
+    - Ubuntu Focal
+    - Ubuntu Bionic
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/reference/daemons/wazuh-authd.html
+    - https://documentation.wazuh.com/current/user-manual/registering/host-verification-registration.html
+
+tags:
+    - enrollment
+'''
+import os
+import ssl
+import time
+from pathlib import Path
+
+import pytest
+
+from wazuh_testing.utils.configuration import get_test_cases_data, load_configuration_template
+from wazuh_testing.tools.socket_controller import SocketController
+from wazuh_testing.tools.certificate_controller import CertificateController
+from wazuh_testing.utils import services
+from wazuh_testing.tools.monitors import file_monitor
+from wazuh_testing.constants.paths.logs import WAZUH_LOG_PATH
+from wazuh_testing.utils.callbacks import generate_callback
+from wazuh_testing.modules.authd.patterns import PREFIX
+
+from . import CONFIGURATIONS_FOLDER_PATH, TEST_CASES_FOLDER_PATH
+
+# Marks
+
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=0), pytest.mark.server]
+
+# Paths
+test_configuration_path = Path(CONFIGURATIONS_FOLDER_PATH, 'config_authd_ssl_certs.yaml')
+test_cases_path = Path(TEST_CASES_FOLDER_PATH, 'cases_authd_ssl_certs.yaml')
+
+# Configurations
+test_configuration, test_metadata, test_cases_ids = get_test_cases_data(test_cases_path)
+test_configuration = load_configuration_template(test_configuration_path, test_configuration, test_metadata)
+
+SSL_AGENT_CA = '/var/ossec/etc/test_rootCA.pem'
+SSL_AGENT_CERT = '/tmp/test_sslagent.cert'
+SSL_AGENT_PRIVATE_KEY = '/tmp/test_sslagent.key'
+
+AGENT_ID = 0
+AGENT_NAME = 'test_agent'
+AGENT_IP = '127.0.0.1'
+WRONG_IP = '10.0.0.240'
+INPUT_MESSAGE = "OSSEC A:'{}_{}'"
+OUPUT_MESSAGE = "OSSEC K:'"
+
+# Simulation options
+# a. Unverified Host:
+# - No certificate
+# - Valid Certificate
+# - Incorrect Certificate
+# b. Verified Host:
+# - No cerificate
+# - No certificate
+# - Valid Certificate
+# - Incorrect Certificate
+# - Valid certificate, Incorrect Host
+# Variables
+
+receiver_sockets_params = [((AGENT_IP, 1515), 'AF_INET', 'SSL_TLSv1_2')]
+
+monitored_sockets_params = [('wazuh-modulesd', None, True), ('wazuh-db', None, True), ('wazuh-authd', None, True)]
+
+receiver_sockets, monitored_sockets = None, None
+
+
+def restart_wazuh_authd():
```

Comment:
Try to use a fixture instead. Also, use as many defines from the framework as possible.